### PR TITLE
nixos/factorio: add openFirewall option

### PR DESCRIPTION
**Pull request by Stekke**
###### Motivation for this change

According to the nixpkgs manual in "20.4. New modules" services should not open ports in the firewall by default .
However the factorio module currently does this, and is preventing me from creating more strict firewall rules.
Worse still, the factorio module offers no option to disable this behavior.

This pr adds the openFirewall option, set to true by default.
Thereby existing setups remain working and this would allows easy backporting of this commit to nixos 20.09.
But people who want to setup more complicated firewall rules can do so, without interference from the factorio module using the openFirewall option.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by Baughn — APPROVED** [review]

File: nixos/modules/services/games/factorio.nix [hunk 1]
```diff
@@ -49,8 +49,13 @@ in
         default = 34197;
         description = ''
           The port to which the service should bind.
-
-          This option will also open up the UDP port in the firewall configuration.
+        '';
+      };
+      openFirewall = mkOption {
+        type = types.bool;
+        default = true;
```
Comment (erictapen) [inline nixos/modules/services/games/factorio.nix:56]: Actually I'd even advocate for defaulting to false here. Imho the only module that should open ports opt-out is openssh. Maybe you could add a second commit defaulting to false, so we can backport only the first commit?

**Review by erictapen — COMMENTED** [review]
Good point, that is really bad behaviour of the factorio module.

File: nixos/modules/services/games/factorio.nix (hunk 1 above)
Comment (Stekke) [reply nixos/modules/services/games/factorio.nix]: Yeah that makes sense.

**Review by erictapen — APPROVED** [review]
Thanks.

**Comment by erictapen** [comment]
Backport: d31c86b12431257c1fa4fbc6e997bd557240b4c8

**Pull request merged**